source/agora/flash/Channel.d

```diff
@@ -2270,6 +2264,56 @@ LOuter: while (1)
         return update_tx;
     }
 
+    /***************************************************************************
+
+        Prepare the settlement TX and publish
+
+        Params:
+            update = update to be published
+
+    ***************************************************************************/
+
+    protected Transaction publishSettlementTx (in UpdatePair update)
+    {
+        auto settle_tx = update.settle_tx.serializeFull.deserializeFull!Transaction();
+        assert(this.last_externalized_update_utxo != Hash.init);
+
+        settle_tx.inputs[0].utxo = this.last_externalized_update_utxo;
+
+        auto utxos = this.getFeeUTXOs(settle_tx.sizeInBytes());
+        settle_tx.inputs ~= utxos.utxos.map!(hash => Input(hash)).array;
+        settle_tx.inputs.sort();
+
+        if (utxos.total_value.sub(utxos.total_fee) && utxos.total_value > 0.coins)
+            settle_tx.outputs[update.multi_settle_sig.output_idx] = Output(utxos.total_value, this.flash_conf.key_pair.address);
+        else
+            // todo: this is a requirement of SigHash.OmitSingle, fix the need to always have a refund output
```

Comment:
Shouldn't we create an issue for this for a better explanation about the problem?

Reply:
Ok, I will raise an issue for it,

Reply:
Did you raise this issue?

Reply:
Just did, https://github.com/bosagora/agora/issues/2753